src/wechaty/user/room.py

```diff
@@ -98,17 +105,20 @@ def __init__(self, room_id: str) -> None:
             )
             raise Exception("Room create error")
 
+
+
     @classmethod
-    async def find_all(cls, query: RoomQueryFilter = None) -> List[Room]:
+    async def find_all(
+            cls,
+            query: Union[str, RoomQueryFilter] = None) -> List[Room]:
         """
         find room by query filter
         :param query:
         :return:
         """
         log.info('Room find_all <%s>', json.dumps(query))
-
-        puppet_query = query.transfer() if query is not None else None
-        room_ids = await cls.get_puppet().room_search(puppet_query)
+        room_query = _build_room_query(query)
+        room_ids = await cls.get_puppet().room_search(room_query)
```

Comment:
We are using a naming style as `room_id_list` instead of `room_ids` because the `_list` postfix can make the variable type more explicit.

So I'd like to suggest let's follow this naming style.

Please feel free to let me know what you think, thanks.

Reply:
I have notice the name style from ts-wechaty. But I sugguest to name list variable with the plural of whatever's in it.
> [stackoverflow](https://stackoverflow.com/questions/7785071/better-python-list-naming-other-than-list/7785095)

**reasons**:
- **_list variable means colletion type, plural also means it.
- **s variable defaultly means list type. eg: names (List[str]),  ids(List[str]).
- **s is shorter than **_list
- **s is more english readable than **_list

Reply:
Ok, I'll not consist of it.

I use `list` instead of the `plural` is because `_list` is more readable and can differentiate with the single variable better.

For example: `room vs rooms` is not as clear as `room vs room_list`.

Reply:
I think the discuss on variable name style is interesting. 

First, I reviewed some open source project code in python just now.  eg: [flask](https://github.com/pallets/flask/blob/master/src/flask/blueprints.py#L143) They almost name the colletion variable with s suffix. I think all developers are familiar with english, and `plural`  words means collection is the basic gramma. So one from gramma aspect, another from word aspect. Two way can be accepted.

Second, `readable` can be a cognitive problems, if a word is rich in meaning, the name is important because it will influence the understanding. But if a word is simple, the name should be simple too.  
```python
# first way
room_list = search_room_list(query)
for room in room_list:
    # do something
# second way
rooms = search_rooms(query)
for room in rooms:
    # do something
```
Personally, I think second way can be **concise**, **shorter** and **simple** 😂😂😂.

Reply:
Good to read your sharing and thoughts.

I agree with you that the variable name with `s` suffix for collections is concise, short and simple. On the other side, the `_list` suffix will be a little verbose but make a strong meaning for the variable: it's a collection.

Could you please create a wiki page with the title **Coding Style** and copy/paste this conversation over there? Because this discussion is very valuable and important.

After creating the Wiki page, we can follow your favorite style (the second way, with `s`) from now and then follow it in all of our Python codebases.

BTW: your link of the `flask` is not a good example for our discussion because it's not a collection variable: it's a boolean type.